main.nf

```diff
@@ -1081,6 +1093,107 @@ process detect_lncRNA {
     """
 }
 
+// Process merge groups for quantification
+// #######################################
+
+if (merge_quantification) {
+
+  // Add maps to merge groups
+  // ########################
+  groups_to_merge_for_quantification = groups_to_merge_for_quantification.toList().get()
+
+  maps_to_merge_for_quantification.map { map ->
+    [
+      groups_to_merge_for_quantification.find { group ->
+        map[0] in group[1]
+      }[0]
+    ] + map
+  }.groupTuple().tap {
+    maps_to_check_before_quantification
+  }.map {
+    [it[0], (it[2].sum()/it[2].size()).toInteger(), it[3][0], it[4]]
+  }.set {
+    maps_to_merge_for_quantification
+  }
+
+  // Check differing read lengths and directions for quantification groups
+  // #####################################################################
+  error = ''
+
+  maps_to_check_before_quantification = maps_to_check_before_quantification.toList().get().sort { a, b -> a[0] <=> b[0] }
+
+  differing_lengths = maps_to_check_before_quantification.findAll {
+    it[2].subsequences().findAll {
+      it.size() == 2
+    }.collect {
+      (it[0] - it[1]).abs()
+    }.findAll {
+      it > 10
+    }.size() > 0
+  }.collectEntries { group ->
+    [
+      (group[0]): group[1].withIndex().collect { it, i ->
+        it + ' (' + group[2][i] + ')'
+      }
+    ]
+  }
+
+  if (differing_lengths.size() > 0) {
+    error += 'Cannot merge differing read lengths for quantification:\n  '
+    error += differing_lengths.collect {
+      it.key + ': ' + it.value.join('  ')
+    }.join('\n  ') + '\n'
+  }
+
+  differing_directions = maps_to_check_before_quantification.findAll {
+    it[3].toUnique().size() > 1
+  }.collectEntries { group ->
+    [
+      (group[0]): group[1].withIndex().collect { it, i ->
+        it + ' (' + group[3][i] + ')'
+      }
+    ]
+  }
+
+  if (differing_directions.size() > 0) {
+    error += 'Cannot merge differing read directions for quantification:\n  '
+    error += differing_directions.collect{ it.key + ': ' + it.value.join('  ') }.join('\n  ') + '\n'
+  }
+
+  if (error) exit 1, error
+
+
+  // Merge maps for quantification
+  // #############################
+  process merge_quantification {
+
+    label 'cpu_16'
+
+    input:
+      tuple val(prefix), val(length), val(direction), path(maps) from maps_to_merge_for_quantification
+
+    output:
+      tuple val(prefix), val(length), val(direction), path('*.bam') into maps_to_quantify
+      path '*.bam' into maps_to_control_exons
+
+    script:
+      """
+      samtools merge "$prefix".bam $maps --threads ${task.cpus}
+      """
+  }
+
+} else {
+  maps_to_merge_for_quantification.tap {
+    maps_to_quantify
+  }.map {
+    [it[0], it[2], it[3]]
```

Comment:
This maps each item to 3 values, but it seems only one value is necessary for controlling exons.